src/js/components/covid19/AmountsVisualization.jsx

```diff
@@ -0,0 +1,858 @@
+    const [outlayValueData, setOutlayValueData] = useState(null);
+    const [outlayLabelData, setOutlayLabelData] = useState(null);
+    const [obligationQuestionData, setObligationQuestionData] = useState(null);
+    const [obligationValueData, setObligationValueData] = useState(null);
+    const [obligationLabelData, setObligationLabelData] = useState(null);
+    const [remainingBalanceCircleData, setRemainingBalanceCircleData] = useState(null);
+    const [zeroPercentData, setZeroPercentData] = useState(null);
+    const [oneHundredPercentData, setOneHundredPercentData] = useState(null);
+    const _totalBudgetAuthorityQuestion = useRef(null);
+    const _totalBudgetAuthorityLabel = useRef(null);
+    const _totalBudgetAuthorityValue = useRef(null);
+    const _obligationQuestion = useRef(null);
+    const _obligationLabel = useRef(null);
+    const _obligationValue = useRef(null);
+    const _outlayQuestion = useRef(null);
+    const _outlayLabel = useRef(null);
+    const _outlayValue = useRef(null);
+    const _remainingBalanceQuestion = useRef(null);
+    const _remainingBalanceLabel = useRef(null);
+    const _remainingBalanceValue = useRef(null);
+    const zerPercentRef = useRef(null);
+    const oneHundredPercentRef = useRef(null);
+    useEffect(() => setLoading(!Object.keys(overviewData).length), [overviewData]);
+    // X Scale
+    useEffect(() => {
+        if (width) {
+            const s = scaleLinear()
+                .domain([0, overviewData._totalBudgetAuthority])
+                .range([amountsPadding.left, width - amountsPadding.right]);
+            setScale(() => s);
+        }
+    }, [width, overviewData]);
+    // totalRectangleData
+    useEffect(() => {
+        if (scale && Object.keys(overviewData).length > 0) {
+            const { offset, fill, text: textInfo } = rectangleMapping._totalBudgetAuthority;
+            const { left, right } = amountsPadding;
+            const amount = Math.abs(overviewData._totalBudgetAuthority);
+            const units = calculateUnits([amount]);
+            const moneyLabel = `${formatMoneyWithPrecision(amount / units.unit, units.precision)} ${units.longLabel}`;
+            const data = {
+                x: left + offset.left,
+                y: startOfChartY + offset.top,
+                width: scale(amount) - (right + (offset.right || 0)),
+                height: rectangleHeight - (2 * offset.bottom),
+                fill,
+                description: `A rectangle with width representative of the ${textInfo.label} amount ${moneyLabel}`
+            };
+            if (!isNaN(scale(amount))) setTotalRectangleData(data);
+        }
+    }, [scale, overviewData]);
+    // outlayRectangleData
+    useEffect(() => {
+        if (scale) {
+            const { offset, fill, text: textInfo } = rectangleMapping._totalOutlays;
+            const { left, right } = amountsPadding;
+            const amount = Math.abs(overviewData._totalOutlays);
```

Comment:
yes, values were showing up negative in the data. I do not know for what reason but Ross said that is a loading issue. Please refer to @tony-sappe or Ross for more information.

Reply:
Most likely from limited mock data. I don't think the `abs()` will cause problems long-term since those values should always be positive with accurate data, but 🤷 .

@BrianZito want to look at the mock data leading to negative values?

Reply:
The values are in fact all negative in DEV. Given the choice between compensating for negative values or just making the fake data follow the rules, I would far prefer the latter. I'll check in with Ross to make sure this is a loading issue we can ignore, and if so I'll just flip all the negative outlays.

Reply:
Barring more info from Ross, I multiplied all the values by negative one. If that's wrong and we need to deal with negative values, we can always just do it again.

Reply:
Update on this: After talking with Ross, it seems that the total_outlays column on our GTAS loader CAN be either negative or positive (based on the data we have), and the loader may have additional requirements we will be needing to add in during later sprints. If we leave this as absolute value, we will likely need to return to this in a later PR. I will leave it to front-end's judgement how to handle this uncertainty.